### DISCUSSION TITLE
Device to Device Communication

### DISCUSSION THREAD

**Discussion by jonaswre** [Luligu/matterbridge#63] [Q&A] [answered]
Hi, I'm thinking about integrating homematic (german smart home vendor). 

One of the main selling points is the device to device communication. If i'm correct zigbee has a similar concept.

The matter specification seems to support that (https://blog.espressif.com/matter-device-to-device-automations-bdbb32365350)

Has anybody tried to implement such a feature. And if so, how could one handle device to device between bridged devices.

**Comment by Luligu** [comment]
Hi,
as for my knowledge, the client server implementation is still only in the specs.
Soon the controllers will hopefully upgrade to Matter 1.3 and maybe...

**Comment by jonaswre** [comment]
Thanks for the answer.

Matterbridge is based on matter.js which in term is a reference implementation right? 

So could I already create a bridge that would allow me to facilitate this communication? Even if it could be configured using a controller yet.

So basically a bridge maps the capabilities of a device to different cluster servers and it needs to be able to take user defined links between two clusters and configure the devices in their protocol. This at least allows device to device behind a bridge.

**Reply by Apollon77** [reply, marked as answer]
In Matter this is called "Binding" and yes it is specified ... but that's it for now.

There is currently no controller that allows to configure bindings and it is completely unknown which end-user devices support what as bindings. The current knowledge is that there might be an Eve device later this year that allows pairing of the eve thermostat with a temperature sensor ( at least there was an announcement for this).

Matter.js is on a way to become the "second reference implementation" if you want to call it that way. The only official reference implementation that exists is the project-chip Matter SDK. But yes the topic on "Cluster clients" and "bindings" is also on the Matter.js todo list, but not yet fully implemented.

So yes mid-term we could have matter.js as device reacting on bindings in a cluster specific way and with cluster specific logic. We could also use matter.js as controller to setup a fabric and configures such bindings. That will work well for all cases where matter.js as device has control over the binding action (e.g. thermostat subscribes to a bound temp sensor to get his value or switch sending toggle command to an onoff device). But as soon as one part of it is a real end-user device that would need to have an active role (like "the switch" is a matter switch device that is expected to send a toggle command to a light) then this might not work.
So once we are there we will see whats needed for it and whats next.

So long story short: Will come, not just now :-)
